jmclient/jmclient/wallet_utils.py

```diff
@@ -894,6 +897,105 @@ def wallet_signmessage(wallet, hdpath, message):
+            except ValueError:
+                jmprint("Invalid choice, must be an integer.", "error")
+                continue
+            if not isinstance(ret, int) or ret < -1 or ret > umax:
+                jmprint("Invalid choice, must be between: -1 and " + \
+                        str(umax) + ", try again.", "error")
+                continue
+            break
+        return ret
+
+    ulist = []
+    # use a counter not 'enumerate' to count over two loops:
+    i = 0
+    for (u, v) in utxos_disabled.items():
+        txid, idx = u
+        value = v['value']
+        jmprint(str(i) + ": " + binascii.hexlify(txid).decode('ascii') + \
+                ":"+str(idx) + "  : " + str(value) + " sats, -- DISABLED")
+        ulist.append((txid, idx))
+        i += 1
+    disabled_max = i
+    for (u, v) in utxos_enabled.items():
+        txid, idx = u
+        value = v['value']
+        jmprint(str(i) + ": " + binascii.hexlify(txid).decode('ascii') + \
+                ":"+str(idx) + " , " + str(value) + " sats, -- ENABLED")
+        ulist.append((txid, idx))
+        i += 1
+    jmprint("Choose an index 0 .. " + str(i-1) + \
+            " to enable/disable or -1 to just quit.")
+    chosen_idx = default_user_choice(i-1)
+    if chosen_idx == -1:
+        return None
+    # the return value 'disable' is the action we are going to take;
+    # so it should be true if the utxos is currently enabled.
+    disable = False if chosen_idx < disabled_max else True
+    return ulist[chosen_idx], disable
+
+def wallet_disableutxo(wallet, md, display_callback=None, info_callback=None):
+    """ Given a wallet and a mixdepth, display to the user
+    the set of available utxos, indexed by integer, and accept a choice
+    of index to disable, then commit this disabling to the wallet storage,
+    so that this disable-ment is persisted. Also allow re-enabling of a
+    chosen utxo which is currently disabled.
+    Callbacks for display and reporting can be specified in the keyword
+    arguments as explained below, otherwise default CLI is used.
+
+    ** display_callback signature:
+    args:
+    1. utxos_enabled ; dict of utxos as format in wallet.py.
+    2. utxos_disabled ; as above, for disabled
+    returns:
+    1.((txid(str), index(int)), disabled(bool)) of chosen utxo
+    for disabling/enabling.
+    ** info_callback signature:
+    args:
+    1. message (str)
+    2. type (str) ("info", "error" etc as per jmprint)
+    returns: None
+    """
+    if display_callback is None:
+        display_callback = display_utxos_for_disable_choice_default
+    if info_callback is None:
+        info_callback = jmprint
+    if md is None:
+        info_callback("Specify the mixdepth with the -m flag", "error")
+        return "Failed"
+    utxos_enabled = wallet.get_utxos_by_mixdepth_()[md]
+    utxos_all = wallet.get_utxos_by_mixdepth_(include_disabled=True)[md]
+    utxos_disabled_keyset = list(set(utxos_all).difference(set(utxos_enabled)))
```

Comment:
why `list()`?

Reply:
Right, I see this is not needed. Done in 1925e0da95a1a1180431a4c3e89a35195f55d889